### PR TITLE
Add auto certificate download feauture and force use external network…

### PR DESCRIPTION
**Pull request by phuonglm**
You could verify this PR with this iorad https://github.com/phuonglm/iorad/tree/features/auto-ssl-download .

### CODE REVIEW

File: vagrant_config.json [hunk 1]
```diff
@@ -186,7 +186,8 @@
             "destination": "/etc/nginx/certs",
             "owner": "vagrant",
             "group": "vagrant",
-            "mode": "0644"
+            "mode": "0644",
+            "auto_certs": "localdev.tk"
```
Comment (hoatle) [inline vagrant_config.json]: should use array instead

File: vagrant_config.json [hunk 2]
```diff
@@ -186,7 +186,8 @@
             "destination": "/etc/nginx/certs",
```
Comment (hoatle) [inline vagrant_config.json:186]: set the sources to be empty array:

```
"sources": [],
````

**Comment by hoatle** [comment]
Whenever I reload --provision, it downloads (renews) the certs. Could we only download/renew when the certs are expired only?

**Review by hoatle — CHANGES_REQUESTED** [review]

**Comment by hoatle** [comment]
for this change, we need to update the docs and the setup with .dev domain, see related changes from https://github.com/teracyhq/dev/pull/426

File: main-cookbooks/teracy-dev/recipes/normalization.rb [hunk 3]
```diff
@@ -38,3 +38,14 @@
     EOH
   not_if { ::File.exist?('/etc/bash_completion') && ::File.exist?('/usr/share/bash-completion/bash_completion') }
 end
+
+bash 'set default route external interface' do
```
Comment (hoatle) [inline main-cookbooks/teracy-dev/recipes/normalization.rb:42]: it's better to move this into another PR for the right issue.

Comment (phuonglm) [reply main-cookbooks/teracy-dev/recipes/normalization.rb:42]: It's is depend, if the VM using host's network connection for certificate download then .... request to localdev.tk will point to VM IP instead of right ip.

Comment (hoatle) [reply main-cookbooks/teracy-dev/recipes/normalization.rb:42]: so get https://github.com/teracyhq/dev/issues/432 done first, then get https://github.com/teracyhq/dev/issues/431 done later.

Comment (hoatle) [reply main-cookbooks/teracy-dev/recipes/normalization.rb:42]: this change will not resolve the problem when I `$ vagrant reload`

**Comment by hoatle** [comment]
closed, will consider this after v0.6.0-a1 release
